README.md
# Mod Readme
Hey So if you got here there must be something that you like or that caught your eye
Nothing big just two snakes that you can spawn in.

**THis Requires the StarboundSimpleVoreMod by sheights** 

found below
https://github.com/Sheights/StarboundSimpleVoreMod

**Install**

Coppy the folder to your mod folder located in the starbound game folder
Do not nest the folder so it should be Starbound/mods/Modfoldernamehere/ not Starbound/mods/Extrafolder/ModFoldername <-bad


**INFO**

They have the same spawn setup: 1 door, 1 Light, 6 floran, 1 Vore badge, Colony deed

They only have the one version and it does not digest.

Each have multi stage swallow anim and allow the user to essentially "escape" the pred.  You can spam interact to get out if needed.
They also allow interaction from outside players to convince the snake to let their prey go.
If you stay too long you will be trapped in the snake until you die (hunger or killed) or convince them to let you out -> this is an instant release.
As you go deeper the chance of escape gets harder :3

Anyway these have been on my hard drive forever and just wanted to upload something to test out the site



changelog.txt
**Version 1.0.1**
-Added Digest versions to the snakes
-Digest begins once the prey reaches the stomach and becomes trapped.  So while you slowly slip down you won't take damage
-Reworked Plea to not completely release prey on success.  Still can release but now has chance to release into different stages of the swallow.
-Lowered time between stageintervals.  Swallow will happen quicker.

-General Program decluttering to make it cleaner and reduce redundant code

**Version 1.0.0**
Base Version Release :D



scripts/vore/npcvoreModified.lua
require "/scripts/vore/exclusions.lua"

---------------------------------------------------------------------------------------
--#####################################################################################
--CONFIG SECTION

npcChance = 1

-- Player Chance is a percent chance that the player will be vored compared to the standard base chance
-- Keep the value anywhere between 0 and 1
-- 0% = 0, 50% = 0.5 or 1/2, 100% chance = 1.0
-- This can also be done in the individual .lua files.
playerChance = 1

-- Audio will activate, Guess what, the audio portion of the mod. This will turn off audio for all preds.
-- For individual preds, you can overide this value to suit your needs.
-- This line can either be "true" or "false"
audio = true

--#####################################################################################
---------------------------------------------------------------------------------------

--Function Preservation

oldInit = init
oldInteract = interact
oldUpdate = update

-----------------------

--Variable Declaration

capacity	= 1
duration 	= 90
reqTimer	= 1
squirmTimer	= 1

speaker		= nil
tempTarget	= nil

isDigest	= false

dead		= {}
isPlayer	= {}
request 	= {}
stopWatch	= {}
victim 		= {}

head		= {}
chest		= {}
legs		= {}
back		= {}

bellyLines		= {}
digestLines		= {}
eatLines		= {}
passLines		= {}
requestLines	= {}

effect = "npcvore"

----------------------

function init()

	for i=0, capacity do
		table.insert(dead, 1, false)
		table.insert(isPlayer, 1, false)
		table.insert(request, 1, false)
		table.insert(stopWatch, 1, 0)
		table.insert(victim, 1, nil)
	end

	head[1] = npc.getItemSlot("head")
	chest[1] = npc.getItemSlot("chest")
	legs[1] = npc.getItemSlot("legs")
	back[1] = npc.getItemSlot("back")
	
	npc.setInteractive(true)	

	oldInit()
	
	initHook()

end

function update(dt)

	tempupdate = update
	tempinteract = interact
	
	if #victim < capacity and isntEaten() and math.random(100) == 1 then
--		sb.logInfo("Attempting to feed")
		feed()
	end
	
	if #victim > 0 then
	
		for i=1, #victim do
			stopWatch[i] = stopWatch[i] + dt
		end
		
		if math.random(500) == 1 and audio then
			world.spawnProjectile( "digestprojectile" , mcontroller.position(), entity.id(), {0, 0}, false )
		end
		
		digest()
	
	end
	
	if reqTimer < 0.7 then
		reqTimer = reqTimer + dt
	end
	
	if squirmTimer < 5 then
		squirmTimer = squirmTimer + dt
	end
	
	oldUpdate(dt)
	updateHook(dt)
	update = tempupdate
	interact = tempinteract
	
end

function interact(args)

--	sb.logInfo("Interracted With")
	if isVictim(args.sourceId) then
--		sb.logInfo("Is a Victim")
		if squirmTimer >= 5 then
			squirmTimer = 0
			speaker = args.sourceId
			squirm(args)
		end
		
		if reqTimer >= 0.7 then
			reqTimer = 0
		else
			reqRelease(args)
			reqTimer = 0.7
		end
	else
		if reqTimer >= 0.7 then
			reqTimer = 0
			oldInteract(args)
		elseif #victim >= capacity then
			rub(args)
		elseif isntEaten() then
			reqFeed(args)
		end
	end
	interactHook(args)
end

function containsPlayer()
	for i=1, #victim do
		if isPlayer[i] then
			return true
		end
	end
	return false
end

function isntEaten()
	tempEffects = status.getPersistentEffects("vore")
	if #tempEffects ~= 0 then
		return false
	end
	tempEffects = status.getPersistentEffects("egg")
	if #tempEffects ~= 0 then
		return false
	end
	return true
end

function digest()

	for i=1, #victim do
	
		--victim[i] is sometimes nil.
		local removeit = false
		if victim[i] ~= nil then
			removeit = not world.entityExists( victim[i] )
		else
			removeit = true;
		end
		if removeit then
			if not world.entityExists( victim[i] ) then
				for j = i, capacity do
					if j == capacity then
						dead[j] = false
						stopWatch[j] = 0
						victim[j] = nil
						request[j] = false
						isPlayer[j] = false
					else
						dead[j] = dead[j+1]
						stopWatch[j] = stopWatch[j+1]
						victim[j] = victim[j+1]
						request[j] = request[j+1]
						isPlayer[j] = isPlayer[j+1]
					end
				end
				do return end
			end
		end
		if isDigest then
			if not dead[i] and world.entityHealth(victim[i])[1] / world.entityHealth(victim[i])[2] < 0.03 then
				dead[i] = true
				deathHook( victim[i] )
			end
		end
		if stopWatch[i] >= duration and request[i] == false then
			world.sendEntityMessage( victim[i], "applyStatusEffect", "voreclear", 1, entity.id() )
			digestHook( victim[i], stopWatch[i], dead[i] )
			for j = i, capacity do
				if j == capacity then
					dead[j] = false
					stopWatch[j] = 0
					victim[j] = nil
					request[j] = false
					isPlayer[j] = false
				else
					dead[j] = dead[j+1]
					stopWatch[j] = stopWatch[j+1]
					victim[j] = victim[j+1]
					request[j] = request[j+1]
					isPlayer[j] = isPlayer[j+1]
				end
			end
		dress()
		end
	end

end

function dress()

	local temphead
	local tempchest
	local templegs
	local tempback
	
	for i = 1, #victim + 1 do
		if head[i] ~= nil then
			temphead = head[i]
		end
		if chest[i] ~= nil then
			tempchest = chest[i]
		end
		if legs[i] ~= nil then
			templegs = legs[i]
		end
		if back[i] ~= nil then
			tempback = back[i]
		end
	end
	
	npc.setItemSlot( "head", temphead )
	npc.setItemSlot( "chest", tempchest )
	npc.setItemSlot( "legs", templegs )
	npc.setItemSlot( "back", tempback )

end

function exclusionCheck (target)

	if world.isNpc(target) then
		for i = 1, #exclusionList do
			if world.npcType(target) == exclusionList[i] then
				return false
			end
		end
	end
	return true

end

function feed()
	--check area for prey
	local people = world.entityQuery( mcontroller.position(), 7, {
		withoutEntityId = entity.id(),
		includedTypes = {"npc", "player"},
		boundMode = "CollisionArea"
	})
	
	if #people == 0 then
--		sb.logInfo("No prey found")
		do return end
	end
--	sb.logInfo("Food Found")
	--check for projectiles
	
	local eggCheck = world.entityQuery( mcontroller.position(), 7, {
		withoutEntityId = entity.id(),
		includedTypes = {"npc", "player", "projectile"},
		boundMode = "CollisionArea"
	})

	if #people ~= #eggCheck then
--		sb.logInfo("Projectile found")
		do return end
	end

	--select a random victim

	tempTarget = people[math.random(#people)]

	--check the exclusions
	
	if not exclusionCheck(tempTarget) then
--		sb.logInfo("Exclusion Found")
		do return end
	end

	--check already eaten
	
	if isVictim(tempTarget) then
		do return end
	end
	
	--check space between them

	local collisionBlocks = world.collisionBlocksAlongLine(mcontroller.position(), world.entityPosition( tempTarget ), {"Null", "Block", "Dynamic"}, 1)
	if #collisionBlocks ~= 0 then
--		sb.logInfo("No line of sight")
		return
	end

	if not world.isNpc(tempTarget)then
		if math.random() < playerChance then
			isPlayer[#victim+1] = true
		else
			do return end
		end
	elseif math.random() > npcChance then
		do return end
	end
	
	--send message

	victim[#victim+1] = tempTarget
	world.sendEntityMessage( tempTarget, "applyStatusEffect", effect, duration, entity.id() )
	spawnSoundProjectile( "swallowprojectile" )

	--adjust states
	dress()
	feedHook()
end

function isVictim(input)

	for i = 1, #victim do
		if victim[i] == input then
			do return true end
		end
	end
	do return false end
end

function reqFeed(input)
	
	victim[#victim + 1] = input.sourceId
	request[#victim] = true
	isPlayer[#victim] = true
	world.sendEntityMessage( victim[#victim], "applyStatusEffect", effect, duration, entity.id() )
	
	dress()
	requestHook(input)
end

function release()  -- Called as release() meaning no variable is passed so it can be used when needed to remove victims.

--	sb.logInfo("Attempting release")
-- No longer checks for an interaction.
	for i=1, #victim do
			world.sendEntityMessage( victim[i], "applyStatusEffect", "voreclear", 1, entity.id() )
			releaseHook(stopWatch[i])
			for j = i, capacity do

				if j == capacity then
					dead[j] = false
					stopWatch[j] = 0
					victim[j] = nil
					request[j] = false
					isPlayer[j] = false
				else
					dead[j] = dead[j+1]
					stopWatch[j] = stopWatch[j+1]
					victim[j] = victim[j+1]
					request[j] = request[j+1]
					isPlayer[j] = isPlayer[j+1]
				end
			end
		dress()
	end
end

function reqRelease(input, time)

--	sb.logInfo("Attempting release")
	for i=1, #victim do
		if victim[i] == input.sourceId then
			world.sendEntityMessage( victim[i], "applyStatusEffect", "voreclear", 1, entity.id() )
			releaseHook(input, stopWatch[i])
			for j = i, capacity do

				if j == capacity then
					dead[j] = false
					stopWatch[j] = 0
					victim[j] = nil
					request[j] = false
					isPlayer[j] = false
				else
					dead[j] = dead[j+1]
					stopWatch[j] = stopWatch[j+1]
					victim[j] = victim[j+1]
					request[j] = request[j+1]
					isPlayer[j] = isPlayer[j+1]
				end
			end
			dress()
		end
	end

end

function spawnSoundProjectile( projectilename )
	if enableAudio then
		world.spawnProjectile( projectilename , mcontroller.position() , entity.id(), {0, 0}, false );
	end
end

function sayLine(input)

	if #input <= 0 then
		do return end
	end

	line = input[math.random(#input)]
	if #victim > 0 then
		line = string.gsub( line, "<entityname>", world.entityName( victim[1] ) )
		npc.say( line )
	elseif not speaker == nil then
		line = string.gsub( line, "<entityname>", world.entityName( speaker ) )
		npc.say( line )
	else
		npc.say( input[math.random(#input)])
	end
end

function squirm(args)
	oldInteract(args)
end

function deathHook(input) end
function digestHook(id, time, dead) end
function feedHook() end
function initHook() end
function interactHook(input) end
function releaseHook(input, time) end
function relHook(time) end
function requestHook(input) end
function rub(input) end
function updateHook(input) end

npcs/vore/KBE/snake/pred_dsnake.lua
require "/scripts/vore/npcvoreModified.lua"
--########################################
--  Based off of the Argo pred in the SSVM
--########################################

struggleFlag = false
exhausetedTimer = 2
animFlag = false
animTimer = 0
isEaten = false
inStomach = false

stages = 7
stageInterval = 15
effect = "npcdigestvore"
buff = "digestionImmunity"
currentLines = "mouth"
playerLines = {}


--Belly Lines (Idle)----------------------------
playerLines["mouth"] = {	"~Mmmmmmmmmm",
							"^blue;*suckles your body*",
							"*The tongue loops around your neck giving a squeeze, letting you know who's in charge*",
							"*you feel the throat muscles trying to pull you down*",
							"^blue;*schluuuuurp*",
							"Hiss...."
							
}
playerLines["belly1"] = {	"~Mmmm, you feel amazing sliding down my throat.",
							"Hopefully you save some energy, my belly could use a nice rub.",
							"You look so nice as a bulge, anyone ever tell you that",
							"That's it let the muscles carry you deeper. ^red;~mmmmmm",
							"Don't worry you'll be heading somewhere special. hehe",
							"Your going to love this.",
							"Maybe I should have coiled you up first.",
							"Enjoy your trip <entityname>, it's a long way down.",
							"It only gets harder from here.",
							"*Rubs bulge with tail*",
							"All mine <entityname>."
}
playerLines["belly2"] = {	"~Mmmm, can't wait to feel you squirming in my stomach.",
							"You look so nice as a bulge.",
							"That's it let the muscles carry you deeper. ^red;~mmmmmm",
							"Hurry up and get to my stomach, your wasting valuable belly rub energy.",
							"I know, first time is so..... ~enjoyable",
							"Better inside, than outside. *hiccup*",
							"Your trips just getting started!",
							"It only gets harder from here, if you struggle."
}
playerLines["belly3"] = {	"Can't wait for belly rubs! ^red;<3",
							"You tasted amazing <entityname>!",
							"Snake bulge is a better look for you. ^red;~mmmmmm",
							"Your going to love my belly <entityname>. I just Know it ^red<3",
							"Enjoy your trip <entityname>, hate to say it's one way!",
							"It only gets more comfortable the deeper you go.",
							"Giving up <entityname>?  Unless you want to go deeper.",
							"*Caresses bulge with tail* Relax.",
							"Being with me won't be all bad trust me!"
}
playerLines["belly4"] = {	"Must feel nice caressed on all sides.",
							"Don't worry your almost there now <entityname>.",
							"*rests on their belly bulge*",
							"I can feel you getting tired, just relax",
							"Just relax and enjoy yourself, I certainly am.",
							"~Mmmmm, don't be afraid to rub around, ~ooooo just like that.",
							"You still awake?",
							"Nothing like live prey ~hehe."
}

playerLines["belly5"] = {	"Now I can really feel you wiggling",
							"You should really give up, it's more fun that way",
							"Almost there now won'r be long till your mine <enitityname>.",
							"~Mmmmmm just just relax, you know you want to.",
							"*hugs their belly* You look adorable.",
							"I can tell your enjoying this."
}

playerLines["belly6"] = {	"Please don't fight it, it's hard convincing people they're not food",
							"Just accept that you are my meal <entityname>.",
							"Don't be afraid to really thrash about, my scales could use a good stretch.",
							"I hope your comfy cause that's going to be your new home <entityname>, forever.",
							"Last chance to accept your place as a belly bulge."
}

playerLines["trapped"] = {	" <entityname> your mine, forever. *lays tail over your bulge*",
							"Like your new home <entityname>, cause your never leaving. ~hehe",
							"I love belly rubs, especially from the inside",
							"You are much better as a squirming bulge <entityname>.",
							"Enjoying yourself?  I know I am. ^red;<3",
							"What's better than being me?  Nothing!",
							"Ohhhhhh~ still a little bit of energy left I see.",
							"~Mmmmmmmmm, just enjoying your squirming.",
							"Relax, it's not like your going anywhere, hehe.",
							"Maybe you wanted this all along <entityname>, to be mine.",
							"Nothing better than squirming prey.",
							"Now your where you belong, as a part of me.",
							"~Hehe you look adorable surrounded by snake scales.",
							"Your better as a part of me.",
							"Admit it, your enjoying this."
}


--Struggle Lines (fail)-------------------------
playerLines["bellyFail"] = {	"There's no point, your going to be mine.",
							"Your squirms aren't helping with your escape.",
							"You look so cute as a squirming bulge. <3",
							"You look so nice as a squirming bulge, luckily you'll be that way forever. <3",
							"~mmmmm Thanks for the belly rub, but you can't get out that easy.",
							"Struggling is pointless.",
							"I can see you trying, but it's not helping.",
							"You tasted to good to let out this easy.",
							"Listen, it's nothing personal but I can't let you go.",
							"That's it just flail around pointlessly, ~mmmmmmm.",
							"Flailing around just makes it easier to enjoy you.",
							"No, you need to be going deeper.  It's more fun that way.",
							"That attempt was pretty pathetic.",
							"Oh, your just to cute as a bulge to let out."
}
playerLines["mouthFail"] = {	"^blue;*suckle* ^white;~mmmphf",
							"*The tongue explores your form intimately distracting you from escape*",
							"^blue;*slurps* ^red;*moan*",
							"~Mmmmm",
							"*teases you with fake swallows letting you know where you'll be heading*",
							"*your shoved back in by the serpent's tail* ^red;hiss.."
}


--Struggle Lines (Success)----------------------
playerLines["bellySucc"] = {	"Hey! I said the trip was one way, <entityname>.",
							"Why are you so determined to not enjoy this.",
							"I suppose I can drag this out a little longer, but I'm still not letting you out.",
							"Your going the wrong way, but I suppose that's only temporary.",
							"Come on and just relax."
}
playerLines["mouthSucc"] = {	"But you tasted so good!",
							"Oh well, another time perhaps?",
							"You only delay the inevitable.",
							"~mmmmm enjoy your freedom little morsel.",
							"Hey, that's not nice",
							"Struggling is pointless, just relax and give in"
}


--Action Lines ---------------------------------
playerLines["pleaFail"] = {	"There's no point, your mine now.",
							"Why should I, you belong to me now <entityname>.",
							"You look so nice as a squirming bulge, luckily you'll be that way forever. <3",
							"~mmmmm Thanks for the belly rub, but I'm still not letting you out.",
							"No one escapes my coils, accept your place.",
							"Struggling is pointless, just relax."
}
playerLines["pleaSucc"] = {	"Mmmmmmmmm~ let's start again shall we",
							"Ok, I'll let you out, but don't think you'll be staying out.",
							"~mmmmm Thanks for the belly rub, I guess I can drag this out a little longer.",
							"I think that message deserves a reward.",
							"I suppose dragging this out isn't such a bad thing. <3"
}

playerLines["convinceSucc"] = { 	"You win, I'll release them.",
									"I suppose I can let them out this one time.",
									"Fine I'll let them out.  Just quit pestering me.",
									"Now that there free.  You want to jump in?"
}

playerLines["convinceFail"] = { 	"I think I'll keep, <entityname> a little longer.",
									"Let me enjoy my meal in peace",
									"Leave me alone unless you want to join them"
}
playerLines["swallow"] = {	"Delicious"
}
playerLines["exhausted"] = {	"you try to catch your breath"
}

function initHook()

	if storage.belly == nil then
		storage.belly = ""
	end
	
	head = {
		name = "snakehead"
	}
	
	chest = {
		name = "snakechest"
	}
	
	legs = {
		name = "snakelegs"
	}

	headbelly1 = {
		name = "snakeheadbelly1"
	}
	headbelly2 = {
		name = "snakeheadbelly2"
	}
	
	--Leg belly
	
	legsbelly1 = {
		name = "snakebellylegs1"
	}
	legsbelly2 = {
		name = "snakebellylegs2"
	}
	legsbelly3 = {
		name = "snakebellylegs3"
	}
	legsbelly4 = {
		name = "snakebellylegs4"
	}
	legsbelly5 = {
		name = "snakebellylegs5"
	}
	legsbelly6 = {
		name = "snakebellylegs6"
	}
	legsbelly7 = {
		name = "snakebellylegs7"
	}
	legsbelly8 = {
		name = "snakebellylegs8"
	}
end

function feedHook()
	victim[#victim + 1] = temptarget
	request[#victim] = true
	isPlayer[#victim] = true
	storage.belly = "1"
	cloth(head, legs, chest)
	animFlag = true
	sayLine(playerLines["swallow"])
	isEaten = true
end

function requestHook(input)
	storage.belly = "1"
	cloth(head, legs, chest)
	animFlag = true
	sayLine(playerLines["swallow"])
	isEaten = true
end

function loseHook()

	npc.say("Is this even called")
	storage.belly = ""
	cloth(head, legs, chest)
	isPlayer = false
	
end

function interact(args)
	
	if reqTimer < 0.7 then
		if #victim > 0 then
			if request[#victim] and args.sourceId == victim[#victim] then --check to make sure request came from victim
				escapeAttempt(args)
			else
				convince()
			end
		else
			victim[#victim + 1] = args.sourceId
			request[#victim] = true
			isPlayer[#victim] = true
			world.sendEntityMessage( victim[#victim], "applyStatusEffect", effect, duration, entity.id() )
			storage.belly = "1"
			dress()
			requestHook(input)
		end
		do return end
	else
		reqTimer = 0
	end
	
	interactHook()
	oldInteract(args)
end

function escapeAttempt(args)
		
		if stopWatch[#victim] > stageInterval*7 then  --Trapped must plea with pred to be freed (won't let you go easy)
			plea(args)
		elseif stopWatch[#victim] > stageInterval*6 and stopWatch[#victim] <= stageInterval*7 then
			escapeRoll(6, args)
		elseif stopWatch[#victim] > stageInterval*5 and stopWatch[#victim] <= stageInterval*6 then
			escapeRoll(5, args)
		elseif stopWatch[#victim] > stageInterval*4 and stopWatch[#victim] <= stageInterval*5 then
			escapeRoll(4, args)
		elseif stopWatch[#victim] > stageInterval*3 and stopWatch[#victim] <= stageInterval*4 then 
			escapeRoll(3, args)
		elseif stopWatch[#victim] > stageInterval*2 and stopWatch[#victim] <= stageInterval*3 then
			escapeRoll(2, args)
		elseif stopWatch[#victim] > stageInterval and stopWatch[#victim] <= stageInterval*2 then 
			escapeRoll(1, args)
		else
			escapeRoll(0, args)
		end
end

function convince()
	if (math.random(200) >= math.random(500)) then
		storage.belly = ""
		release()
		cloth(head, legs, chest)
		sayLine(playerLines["convinceS"])
	else
		sayLine(playerLines["convinceF"])
	end
	
end

function plea(args)
	if (math.random(100) >= math.random(800)) then
		sayLine(playerLines["pleaSucc"])
		stopWatch[#victim] = stopWatch[#victim] -(stageInterval/2 + math.random(stageInterval * stages))
		if (stopWatch[#victim] < 0) then	
		storage.belly = ""
		reqRelease(args)
		cloth(head, legs, chest)
		end
		
	else
		sayLine(playerLines["pleaFail"])
	end
	
end

function escapeRoll(stage, args)
			if (math.random(200) >= math.random(225 + (25 *stage))) then
				if stage == 0 then
				storage.belly = ""
				reqRelease(args)
				cloth(head, legs, chest)
				sayLine(playerLines["mouthSucc"])
				else
				stopWatch[#victim] = stopWatch[#victim] - stageInterval
				sayLine(playerLines["bellySucc"])
				end
			else
				if stage == 0 then
				sayLine(playerLines["mouthFail"])
				else
				sayLine(playerLines["bellyFail"])
				end
			end
end

function cloth(pHead, pLegs, pChest)
		npc.setItemSlot( "head", pHead)
		npc.setItemSlot( "chest", pChest)
		npc.setItemSlot( "legs", pLegs)
end


function digestingCheck()

	if (isEaten and not inStomach) then
		world.sendEntityMessage( victim[#victim], "applyStatusEffect", buff, stageInterval, entity.id() )
		inStomach = true
	end
	
end

function digestBellyChange()
	if #victim ~= null then
		health = world.entityHealth(victim[#victim])[1] / world.entityHealth(victim[#victim])[2]
		if health < .02 then
			release()
			cloth(head, legs, chest)
		elseif health < .10 then
			cloth(head, legs, chest)
		elseif health < .35 then
			cloth(head, legsbelly8, chest)
		elseif health < .60 then
			cloth(head, legsbelly7, chest)
		else
			cloth(head, legsbelly6, chest)
		end
	end
end

function updateHook(dt)
		
	if animFlag then 
		if animTimer < 0.7 then
			npc.setItemSlot( "head", headbelly1 )
		elseif animTimer < 1.7 then
			npc.setItemSlot( "head", headbelly2 )
		else
			animFlag = false
			animTimer = 0
		end
		animTimer = animTimer + dt
	else
		if #victim == 1 then
			if stopWatch[#victim] > stageInterval*7.5 then
				if containsPlayer() then
					stopWatch[#victim] = stageInterval*7.4
				else
					release()
					cloth(head, legs, chest)
				end
			end
			if stopWatch[#victim] > stageInterval*7 then
				digestBellyChange()
				storage.belly = "8"
				currentLines = "trapped"
				inStomach = true
			elseif stopWatch[#victim] > stageInterval*6 then
				cloth(head, legsbelly6, chest)
				storage.belly = "7"
				currentLines = "belly6"
				inStomach = true
			elseif stopWatch[#victim] > stageInterval*5 then  
				cloth(head, legsbelly5, chest)
				storage.belly = "6"
				currentLines = "belly5"
				inStomach = false
			elseif stopWatch[#victim] > stageInterval*4 then
				cloth(head, legsbelly4, chest)
				storage.belly = "5"
				currentLines = "belly4"
				inStomach = false
			elseif stopWatch[#victim] > stageInterval*3 then 
				cloth(head, legsbelly3, chest)
				storage.belly = "4"
				currentLines = "belly3"
				inStomach = false
			elseif stopWatch[#victim] > stageInterval*2 then 
				cloth(head, legsbelly2, chest)
				storage.belly = "3"
				inStomach = false
				currentLines = "belly2"
			elseif stopWatch[#victim] > stageInterval then
				cloth(head, legsbelly1, chest)
				storage.belly = "2"
				inStomach = false
				currentLines = "belly1"
			elseif stopWatch[#victim] <= stageInterval then
				cloth(headbelly2, legs, chest)
				storage.belly = "1"
				inStomach = false
				currentLines = "mouth"
			end
		end
	end
	
	digestingCheck()
	
	if containsPlayer() and math.random(700) < 10 then
			sayLine( playerLines[currentLines] )
		
	end
end
